src/pipeline.py

```diff
@@ -0,0 +1,78 @@
+import multiprocessing as mp
+import os
+import shutil
+
+from common import get_files, get_log
+from extract import extract
+from ingest import ingest
+from parse import parse
+from standardize import standardize
+
+log = get_log(__file__)
+
+
+def clean_dirs(list_of_dirs):
+    """
+    Deletes and recreates empty directories
+    """
+    for dir_ in list_of_dirs:
+        try:
+            shutil.rmtree(dir_)
+        except FileNotFoundError:
+            pass
+        os.makedirs(dir_)
+
+
+def get_filename_without_extension(filename):
+    return os.path.basename(filename).split(".")[0]
+
+
+def get_pipeline_files(raw_dir, extracted_dir, parsed_dir, standardized_dir):
+    """
+    yields file names corresponding to the raw, extracted, parsed, standardized
+    intermediate steps pf the pipeline
+    """
+    suffix = ".json"
+    for raw_file in get_files(raw_dir):
+        filestem = get_filename_without_extension(raw_file)
+        extracted_file = os.path.join(extracted_dir, filestem + suffix)
+        parsed_file = os.path.join(parsed_dir, filestem + suffix)
+        standardized_file = os.path.join(standardized_dir, filestem + suffix)
+        yield raw_file, extracted_file, parsed_file, standardized_file
+
+
+def _etl(raw, extracted, parsed, standardized):
```

Comment:
this is probably an unnecessary comment / feature request at this stage, but we don't check to see if extract / parse /standardize finished successfully before proceeding to the next step, do we?

Reply:
I think an exception in any of extract, parse, or standardize will prevent the ingestion from running (old data won't be overridden). But if no exception happens then we don't check that nothing weird happened, I'm not sure what such a check would look like.